### PR TITLE
feat: Add pre-commit checks

### PR DESCRIPTION
**Pull request by maxxgx**
Adds repository-wide pre-commit checks with `prek` and updates the codebase to pass them cleanly.

## Changes

- add `.pre-commit-config.yaml` and `.markdownlint.yaml`
- enable `ruff`, `pyrefly`, `uv lock`, `hadolint`, `markdownlint`, and `zizmor`
- update `pyproject.toml` and `uv.lock` for the new tooling
- refactor SO101 code by extracting calibration and constants modules
- clean up SO101 implementation, verification, tests, examples, and docs to satisfy linting, formatting, and type checks

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2025 Intel Corporation
```
Comment (samet-akcay) [inline .pre-commit-config.yaml]:
```suggestion
# Copyright (C) 2026 Intel Corporation
```

File: pyproject.toml [hunk 2]
```diff
@@ -97,7 +101,10 @@ extend-exclude = ["docs", "examples", "tests", "notebooks"]
 
 [tool.ruff.lint]
 select = ["ALL"]
-ignore = []
+ignore = [
+    "COM812",  # Conflicts with Ruff formatter
```
Comment (samet-akcay) [inline pyproject.toml:105]: is this a bad thing, why ignore?

File: pyproject.toml [hunk 3]
```diff
@@ -97,7 +101,10 @@ extend-exclude = ["docs", "examples", "tests", "notebooks"]
...
+    "UP037",   # Removes quotes inside Literal[], breaking string literal values
```
Comment (samet-akcay) [inline pyproject.toml]: can we use noqa instead for this edge case, instead of completely ignoring this rule ?

File: pyproject.toml (hunk 2 above)
Comment (maxxgx) [reply pyproject.toml:105]: COM812 is redundant/conflicts when Ruff formatter is enabled, and Ruff recommends disabling it in formatter-driven setups

File: pyproject.toml (hunk 3 above)
Comment (maxxgx) [reply pyproject.toml]: Good point. I’ll replace the project-wide `UP037` ignore with a `noqa` on the affected lines

Comment (maxxgx) [reply pyproject.toml]: It seems `noqa` is not actually needed. Ruff removes the quotes only when you import `from __future__ import annotations`

Comment (samet-akcay) [reply pyproject.toml]: that's even better

**Review by samet-akcay — APPROVED** [review]
